### PR TITLE
Screwdriver Pen!

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Adds a pen with an extendable screwdriver bit as a maintenance spawn.

## Why It's Good For The Game
Gives more use for the pen slot, acts as an emergency screwdriver you can just whip out at any time

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

spen_floor

spen_floor_out

spen_use

</details>

## Changelog
:cl:
add: Added screwdriver pen
add: Added s-pen to maint spawns
/:cl:

### CODE REVIEW

**Comment by PestoVerde322** [comment]
Actually a really neat maint loot spawn, I like it. Only complain is that it looks too close to a normal pen, so it could maybe have some extra texture compared to the normal pen, and also, does it do the same stab damage as a Pen or a Screwdriver?

**Comment by etherware-novice** [comment]
currently just pen damage but it should have the eyestab trait

**Comment by DatBoiTim** [comment]
> Gives more use for the pen slot, acts as an emergency screwdriver you can just whip out at any time

Should probably keep pen damage. I've had things like these and they're flimsy.

**Comment by etherware-novice** [comment]
will do that

**Comment by EvilDragonfiend** [comment]
10% isn't that noticeable downside. it should be 20% at least if you want to make people feel the pendriver a real downside.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -1003,6 +1003,11 @@ GLOBAL_LIST_EMPTY(PDAs)
 		if(inserted_item)
 			to_chat(user, "<span class='warning'>There is already \a [inserted_item] in \the [src]!</span>")
 		else
+			if(istype(C, /obj/item/pen/screwdriver))
+				var/obj/item/pen/screwdriver/temp = C
+				if(temp.extended)
+					to_chat(user, "<span class='warning'>You can't fit the pen in!</span>")
+					return
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/PDA/PDA.dm]: I dislike checking for specific types when doing general behaviours like this, stuff specific to an item should be contained within that item.
Perhaps some kind of var on the pen, or checking the size of the item and changing the item size when you extend the screwdriver would improve code quality

File: code/modules/paperwork/pen.dm [hunk 2]
```diff
@@ -238,3 +238,55 @@
...
+	if(user.zone_selected != BODY_ZONE_PRECISE_EYES && user.zone_selected != BODY_ZONE_HEAD)
+		return ..()
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
+		to_chat(user, "<span class='warning'>You don't want to harm [M]!</span>")
+		return
+	if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
+		M = user
+	return eyestab(M,user)
+
+/obj/item/pen/screwdriver/update_icon()
```
Comment (PowerfulBacon) [inline code/modules/paperwork/pen.dm:289]: Don't update force in update_icon, this should be only changing the icon

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 3]
```diff
@@ -1003,6 +1003,9 @@ GLOBAL_LIST_EMPTY(PDAs)
 		if(inserted_item)
 			to_chat(user, "<span class='warning'>There is already \a [inserted_item] in \the [src]!</span>")
 		else
+			if(C.w_class > WEIGHT_CLASS_TINY)
+				to_chat(user, "<span class='warning'>You can't fit the pen in!</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/PDA/PDA.dm]: Should be something along the lines of "The [pen] doesn't fit!"

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 3 above)
Comment (etherware-novice) [reply code/game/objects/items/devices/PDA/PDA.dm]: @PowerfulBacon does just casting [C] to a string get the display name?

**Review by PowerfulBacon — APPROVED** [review]

**Comment by etherware-novice** [comment]
woo!!

**Pull request merged**